### ISSUE TITLE
Using `mod` not `crate`

### ISSUE THREAD

**Issue by yanganto** [kuutamolabs/lightning-knd#401]
Redundant crates will easily making dependency issues in the future.

- `api` crate is only used in `kld` crate.  There is no reason to split into another crate.

- `settings` crate is used in `kld` and `test-utils`, and there is no reason to split `test-utils` into another crate. There is `[dev-dependency]` for test in Rust, we do not need `test-utils`

**Comment by JosephGoulden** [comment]
I was expecting the API crate would be published and used by other rust apps to communicate with the node. I only recently learned about OpenAPI and the potential ability to generate rust code from a specification document. So this is maybe a better way.

I will happily move test-utils and settings into kld if possible.

**Comment by yanganto** [comment]
Rust crate can have `[lib]` and `[bin]` at the same time.  Some really legacy crates will have issues before, and we do not have that.

The case using crates is 
- you have more than one app, and one is only dependent on `api`, and another is dependent on `kld`

Even in the using case above, you still can use the feature to import a crate with features. (This is also the answer for test-utils part)

If things are in the same repo with the same maintainers, it is good to put it in the same crate, then use `mod` and `features`, else you will easily run into issues as you mentioned in #402.

**Comment by JosephGoulden** [comment]
I agree and would like to simplify into one crate I just don't think it is possible without making changes to the organisation of the tests or making some other compromise (due to limitations of the cargo test framework). But if you think you can combine settings and test-utils into kld without affecting the tests then please give it a go.

But please leave the API crate alone, I am going to try to generate it from a swagger spec.

**Comment by yanganto** [comment]
The `setting` crate is fixed in #402
